service-front/module/Application/src/Helpers/LpaFormHelper.php

```diff
@@ -0,0 +1,263 @@
+use Application\Services\SiriusApiService;
+use Laminas\Form\FormInterface;
+use Laminas\Stdlib\Parameters;
+use Laminas\Http\Response;
+
+class LpaFormHelper
+{
+    public function findLpa(
+        string $uuid,
+        Parameters $formData,
+        FormInterface $form,
+        array $siriusCheck,
+        array $detailsData,
+    ): LpaFormHelperResponseDto {
+        $form->setData($formData);
+        $result = [];
+
+        if ($form->isValid()) {
+            if (
+                ! array_key_exists('opg.poas.lpastore', $siriusCheck) ||
+                empty($siriusCheck['opg.poas.lpastore'])
+            ) {
+                $result = [
+                    'status' => 'Not Found',
+                    'message' => "No LPA found.",
+                ];
+                return new LpaFormHelperResponseDto(
+                    $uuid,
+                    $form,
+                    [
+                        'lpa_response' => $result
+                    ],
+                );
+            }
+
+            if (! $this->checkLpaNotAdded($form->get('lpa')->getValue(), $detailsData)) {
+                $result = [
+                    'status' => 'Already added',
+                    'message' => "This LPA has already been added to this ID check.",
+                ];
+                return new LpaFormHelperResponseDto(
+                    $uuid,
+                    $form,
+                    [
+                        'lpa_response' => $result
+                    ],
+                );
+            }
+
+            $idCheck = $this->compareCpRecords($detailsData, $siriusCheck);
+            $statusCheck = $this->checkStatus($siriusCheck);
+            $channelCheck = $this->checkChannel($siriusCheck);
+
+            if ($statusCheck['error'] === true) {
+                $result['status'] = $statusCheck['status'];
+                $result['message'] = $statusCheck['message'];
+            } elseif ($idCheck['error'] === true) {
+                $result['status'] = 'no match';
+                $result['message'] = $idCheck['message'];
+                $result['additional_data'] = [
+                    'name' => $idCheck['name'],
+                    'address' => $idCheck['address'],
+                    'name_match' => $idCheck['name_match'],
+                    'address_match' => $idCheck['address_match'],
+                    'error' => $idCheck['error']
+                ];
+                $result['data'] = [
+                    "case_uuid" => $uuid,
+                    "LPA_Number" => $form->get('lpa')->getValue(),
+                    "Type_Of_LPA" => $this->getLpaTypeFromSiriusResponse($siriusCheck),
+                    "Donor" => $this->getDonorNameFromSiriusResponse($siriusCheck),
+                    "Status" => $statusCheck['status'],
+                    "CP_Name" => $detailsData['firstName'] . " " . $detailsData['lastName'],
+                    "CP_Address" => $detailsData['address']
+                ];
```

Comment:
I'm not really clear what the difference is between "data" and "additional_data". Could `LpaFormHelperResponseDto` be expanded to have additional OOP structure for this information, rather than using an unstructured array?

Reply:
Additional data is when an LPA turns out to have a name or address mismatch, so those details need to be displayed alongside the details for the CP that's identified in the flow. Good idea re the DTO, I'll update it to include that